Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/ParentsOperation.php

```diff
@@ -65,26 +69,28 @@ public function canEvaluate($context)
      */
     public function evaluate(FlowQuery $flowQuery, array $arguments)
     {
-        $parents = [];
+        $parents = Nodes::createEmpty();
+        $findAncestorNodesFilter = FindAncestorNodesFilter::create(
+            NodeTypeConstraints::create(
+                NodeTypeNames::createEmpty(),
+                NodeTypeNames::fromArray(
+                    [NodeTypeName::fromString('Neos.ContentRepository:Root')]
+                )
+            )
+        );
+
         /* @var Node $contextNode */
         foreach ($flowQuery->getContext() as $contextNode) {
-            $node = $contextNode;
-            do {
-                $node = $this->contentRepositoryRegistry->subgraphForNode($node)
-                    ->findParentNode($node->nodeAggregateId);
-                if ($node === null) {
-                    // no parent found
-                    break;
-                }
-                // stop at sites
-                if ($node->nodeTypeName === NodeTypeName::fromString('Neos.Neos:Sites')) {
-                    break;
-                }
-                $parents[] = $node;
-            } while (true);
+            $ancestorNodes = $this->contentRepositoryRegistry
+                ->subgraphForNode($contextNode)
+                ->findAncestorNodes(
+                    $contextNode->nodeAggregateId,
+                    $findAncestorNodesFilter
+                );
+            $parents = $parents->merge($ancestorNodes);
         }
 
-        $flowQuery->setContext($parents);
+        $flowQuery->setContext(iterator_to_array($parents));
```

Comment:
Why does it have to be an array?

Reply:
because flowQueryContext always is an array atm